server/market/order/order.go

```diff
@@ -0,0 +1,326 @@
+func (o *MarketOrder) ID() OrderID {
+	return calcOrderID(o)
+}
+
+// UID computes the order ID, returning the string representation.
+func (o *MarketOrder) UID() string {
+	return o.ID().String()
+}
+
+// SerializeSize returns the length of the serialized MarketOrder.
+func (o *MarketOrder) SerializeSize() int {
+	// Compute the size of the serialized UTXOs.
+	var utxosSize int
+	for _, u := range o.UTXOs {
+		utxosSize += u.SerializeSize()
+		// TODO: ensure all UTXOs have the same size, indicating the same asset?
+	}
+	// The serialized order includes a byte for UTXO count, but this is implicit
+	// in UTXO slice length.
+	return o.Prefix.SerializeSize() + 1 + utxosSize + 1 + 8 + len(o.Address)
+}
+
+// Serialize marshals the MarketOrder into a []byte.
+func (o *MarketOrder) Serialize() []byte {
+	b := make([]byte, o.SerializeSize())
+
+	// Prefix
+	copy(b[:PrefixLen], o.Prefix.Serialize())
+	offset := PrefixLen
+
+	// UTXO count
+	b[offset] = uint8(len(o.UTXOs))
+	offset++
+
+	// UTXO data
+	for _, u := range o.UTXOs {
+		utxoSize := u.SerializeSize()
+		copy(b[offset:offset+utxoSize], u.Serialize())
+		offset += utxoSize
+	}
+
+	// order side
+	var side uint8
+	if o.Sell {
+		side = 1
+	}
+	b[offset] = side
+	offset++
+
+	// order quantity
+	binary.LittleEndian.PutUint64(b[offset:offset+8], o.Quantity)
+	offset += 8
+
+	// client address for received funds
+	copy(b[offset:offset+len(o.Address)], []byte(o.Address))
+	return b
+}
+
+// Type returns MarketOrderType for a MarketOrder.
+func (o *MarketOrder) Type() OrderType {
+	return MarketOrderType
+}
+
+// Remaining returns the remaining order amount.
+func (o *MarketOrder) Remaining() uint64 {
+	return o.Quantity - o.Filled
+}
+
+// Ensure MarketOrder is an Order.
+var _ Order = (*MarketOrder)(nil)
+
+// LimitOrder defines a limit order in terms of a MarketOrder and limit-specific
+// data including rate (price) and time in force.
+type LimitOrder struct {
+	MarketOrder         // order type in the prefix is the only difference
+	Rate        float64 // price as quote asset per base asset
+	Force       TimeInForce
+}
+
+// ID computes the order ID.
+func (o *LimitOrder) ID() OrderID {
+	return calcOrderID(o)
+}
+
+// UID computes the order ID, returning the string representation.
+func (o *LimitOrder) UID() string {
+	return o.ID().String()
+}
+
+// SerializeSize returns the length of the serialized LimitOrder.
+func (o *LimitOrder) SerializeSize() int {
+	return o.MarketOrder.SerializeSize() + 8 + 1
+}
+
+// Serialize marshals the LimitOrder into a []byte.
+func (o *LimitOrder) Serialize() []byte {
+	b := make([]byte, o.SerializeSize())
+	// Prefix and data common with MarketOrder
+	offset := o.MarketOrder.SerializeSize()
+	copy(b[:offset], o.MarketOrder.Serialize())
+
+	// Price rate
+	// var fb bytes.Buffer
+	// _ = binary.Write(&fb, binary.LittleEndian, o.Rate)
+	// copy(b[mSz:], fb.Bytes())
+	binary.LittleEndian.PutUint64(b[offset:offset+8], math.Float64bits(o.Rate))
```

Comment:
Will need to mention IEEE 754 encoding in the spec.

Reply:
All the price rate details needed changes too.  https://github.com/decred/dcrdex/pull/27  As I note in that PR, I don't feel great about the solution yet, but it will at least be workable this way I believe.

Reply:
Integer rate implemented in https://github.com/decred/dcrdex/pull/15/commits/9b17f17dba8d2df6c226df1040c3a73006a00dd6.  Please have a look.  I'm still not 100% certain about this approach, but it's a detail we can sort out as we go.

Reply:
That'll work.